src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -177,4 +172,16 @@ private function validateKey($key)
 
         return $key;
     }
+
+    private function createGenerator(array $keys)
+    {
+        $f = $this->createCacheItem;
+        $now = time();
```

Comment:
you need to call time() in the loop now since you don't know how many time will be spent between iterations